### ISSUE TITLE
FATAL failed to fetch Terraform Variables

### ISSUE THREAD

**Issue by wgordon17** [openshift/installer#1341]
# Version

```console
$ openshift-install version
./openshift-install v0.13.0
```

# Platform (aws|libvirt|openstack):

aws

# What happened?

Trying to install a normal cluster

```
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to fetch dependency of "Bootstrap Ignition Config": failed to fetch dependency of "Master Machines": failed to generate asset "Platform Credentials Check": validate AWS credentials: checking install permissions: error simulating policy: InvalidInput: Invalid Entity Arn: arn:aws:iam::629075485798:root does not clearly define entity type and name.
status code: 400, request id: 79a805bc-3b8e-11e9-be3d-9f977a7e345d
````

# What you expected to happen?

Start installation

# How to reproduce it (as minimally and precisely as possible)?

`./openshift-install create cluster --dir=test`
Everytime

**Comment by wking** [comment]
@markito reported this the other day.  It seems like these checks don't work with the account's root user (which your `...:root` ARN suggests is what you're doing too), although I'm not clear on why.  Maybe there's a way we could check for that case explicitly?  It sounds like there are also [issues with using this API with assumed roles][1].  @joelddiaz, do you see a way to handle these cases?

[1]: https://stackoverflow.com/questions/52941478/simulate-principal-policy-using-assumed-role

**Comment by wgordon17** [comment]
This is just on 0.13.0 too, downgrading to 0.12.0 works just fine 🤷🏻‍♂️

**Comment by joelddiaz** [comment]
i haven't forgotten about this. i've finally got some access keys for a root user to get this reproduced.

i'll see what can be done to detect this and give a more useful message, but to be clear, using the root creds is not recommended https://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html

**Comment by joelddiaz** [comment]
@wking are we okay erroring out when we detect using the 'root' AWS user (with some 'go generate yourself an IAM user' message to go with it)?

what i've found is that the userid for the root user is set to the account number.

**Comment by wking** [comment]
How about printing a warning:

> The AWS account root user is not recommended: https://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html

but then allowing creation to continue?

**Comment by joelddiaz** [comment]
This is what i have in mind https://github.com/openshift/cloud-credential-operator/pull/43
Need to test this out on a live cluster and see how things go.

**Comment by wking** [comment]
Fixed via #1448.
